### PR TITLE
Add TSHttpTxnNextHopPortGet, add NEXT-HOP to header rewrite

### PR DESCRIPTION
**Pull request by traeak**
This adds TSHttpTxnNextHopPortGet in addition to TSHttpTxnNextHopNameGet and adds to header_rewrite as %{NEXT-HOP:HOST} and %{NEXT-HOP:PORT} for SEND_REQUEST_HDR_HOOK.

### CODE REVIEW

**Review by shukitchan — CHANGES_REQUESTED** [review]
I think you are still working on this. 
But please update the plugin documentation on the new changes as well. Thanks.

**Comment by traeak** [comment]
[approve ci rocky]

**Comment by traeak** [comment]
> I think you are still working on this. But please update the plugin documentation on the new changes as well. Thanks.

it was just wip.  should be documented and everything.

File: doc/developer-guide/api/functions/TSHttpTxnNextHopPortGet.en.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+***********************
+
+Synopsis
+========
+
+.. code-block:: cpp
+
+    #include <ts/ts.h>
+
+.. function:: const char *TSHttpTxnNextHopPortGet(TSHttpTxn txnp)
```
Comment (shukitchan) [inline doc/developer-guide/api/functions/TSHttpTxnNextHopPortGet.en.rst:31]: The return type is int instead. A typo here?

**Review by shukitchan — CHANGES_REQUESTED** [review]
Also, I think we need to update documentation for header rewrite plugin as well.

**Review by shukitchan — APPROVED** [review]

**Comment by zwoop** [comment]
Slightly unorthodox to include this in a patch release, but alas, we're not doing 9.3.0 release, and this seems generally useful and safe.

**Pull request merged**